공통 컴포넌트 버튼 제작

ingong opened the pull request:
## 😀 제목

공통 컴포넌트 중 버튼 제작

## ⛅️ 내용

- 공통 컴포넌트 제작
- props 로 category, size, onClick, disabled(optional) 전달
- figma 에서 설정한 컴포넌트 기반으로 버튼 제작

## 🎸특이사항
현민님 리뷰 후
- App 내에서 잘못 Button 을 넣어줬어서, 해당 내용 제거
- Category, Size Type 추가

ingong committed:
feat: 기본 버튼 컴포넌트 추가 [#24]

ingong committed:
feat: props 값에 따른 크기 및 색상 변경 버튼 컴포넌트 생성 [#24]

ingong committed:
refactor: size 에 따른 버튼 스타일 변경 코드 수정 [#24]

ingong committed:
refactor: category 에 따른 색상 및 효과 반영 [#24]

ingong committed:
feat: onClick 이벤트 Type 추가 [#24]

onClick Props 로 함수를 전달하고, 이를 작동할 수 있음.
부모로부터 함수를 전달받고 이 함수가 수행되는 공통 컴포넌트 작성

ingong committed:
chore: Button 설명 주석 추가 [#24]

사용 방법을 명시하는 주석을 추가함
팀원들이 어떤 Props 로 전달하는지 파악할 수 있다.
이후에 추가로 문서화 작업을 진행할 예정이다.

rudy3091 commented on client/src/App.tsx:
```diff
@@ -1,11 +1,16 @@
     <>
       <GlobalStyle />
-      <div>Hello World</div>
+      <div>
+        <Button variant="default" size="small">
+          버튼
+        </Button>
+        <p>Hello World</p>
+      </div>
```
앗.. 우리 App 수정사항은 빼기로 했었지 않나요ㅋㅋㅋㅋㅋ

ingong replied:
수정했습니당ㅎㅎㅎ

rudy3091 commented on client/src/lib/design/Button.tsx:
```diff
@@ -0,0 +1,62 @@
+
+const StyledButton = styled.button<LayoutProps>`
+  margin: 0;
+  border: none;
+  cursor: pointer;
```
이부분은 GlobalStyle에 이미 선언되어 있어서 굳이 안써주셔도 될것같습니다!

ingong replied:
넵ㅎㅎ 수정했어요!

rudy3091 commented on client/src/lib/design/Button.tsx:
```diff
@@ -0,0 +1,62 @@
+`;
+
+/**
+ * category : default | confirm | cancel
+ * size : small | large
```
이런 경우는 union type 으로 선언해주셔도 좋을것 같아요. 만약 저 type들 안에 들어가지 않는 문자열이 전달되면 vscode가 잡아주기 때문에 더 안정적입니다!

``` typescript
type Category = 'default' | 'confirm' | 'cancel'
type Size = 'small' | 'large'
```

ingong replied:
저 방법 궁금했었는데, type 으로 선언하면 되는군요! 수정했습니다! 감사합니다.

rudy3091 reviewed (commented):
고생많으셨습니다! 내일도 고생해봐요 우리

ingong committed:
fix: App 컴포넌트 버튼 제거 [#24]

ingong committed:
refactor: Category, Size Type alias 선언 [#24]

haesoo9410 commented on client/src/lib/design/Button.tsx:
```diff
@@ -0,0 +1,63 @@
+
+  padding: ${(props) => (props.size === 'large' ? '13px 50px' : '10px 15px')};
+  font: ${({ theme }) => theme.font.bold_regular};
+  font-size: ${(props) => (props.size === 'large' ? '16px' : '14px')};
```
regular 보다 작은 폰트를 추가하는 것에 대해 같이 고려해봐야겠네요..

ingong replied:
매의 눈이 시네요...ㅎ 내일 같이 고려해보면 좋을 것 같아요!

ingong committed:
refactor: 코드 리뷰 사항 반영 [#24]

- interface 와 type 간격 띄우기
- active 반복되는 로직 제거
- 버튼 폰트 스타일 : bold_small, bold_medium 적용

ingong committed:
chore: jsDoc 추가 [#24]

Button 컴포넌트 사용관련한 주석 내용 추가
팀원들이 주석을 보고 바로 버튼 컴포넌트를 재사용하기 위함

ingong committed:
chore: 버튼 컴포넌트에 대한 jsdoc 추가 [#24]

팀원들이 주석을 통해 버튼 컴포넌트의 전달 Props 와 type 을 확인할 수 있도록 작성

ingong committed:
chore: origin 과 local 동기화

ingong committed:
Merge branch 'develop' into feature/button

ingong committed:
Merge branch 'develop' into feature/button
